clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/IsolatedIngestionProcessStats.java

```diff
@@ -0,0 +1,85 @@
+package com.linkedin.davinci.stats;
+
+import com.linkedin.venice.stats.AbstractVeniceStats;
+import com.linkedin.venice.stats.Gauge;
+import com.linkedin.venice.utils.RedundantExceptionFilter;
+import io.tehuti.metrics.MetricsRepository;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * IsolatedIngestionProcessStats is a metrics collecting class that aims to collect metrics from isolated ingestion process.
+ * It maintains a map of String to Double to keep the collected metrics from forked process. updateMetricMap() will be called
+ * by a dedicated thread which collects metrics from child process at a fixed rate.
+ * Metrics collected will be registered to main process's MetricRepository and be reported to metrics reporters(like InGraph).
+ * In order to distinguish the collected metrics from the main process metrics, prefix "ingestion_isolation" is added to every metric names.
+ */
+public class IsolatedIngestionProcessStats extends AbstractVeniceStats {
+  private static final Logger LOGGER = LogManager.getLogger(IsolatedIngestionProcessStats.class);
+  private static final RedundantExceptionFilter REDUNDANT_EXCEPTION_FILTER =
+      RedundantExceptionFilter.getRedundantExceptionFilter();
+  private static final String METRIC_PREFIX = "ingestion_isolation";
+
+  private final Map<String, Double> metricValueMap = new HashMap<>();
+
+  public IsolatedIngestionProcessStats(MetricsRepository metricsRepository) {
+    super(metricsRepository, METRIC_PREFIX);
+  }
+
+  public void updateMetricMap(Map<CharSequence, Double> updateMetricValueMap) {
```

Comment:
This strategy potentially doesn't work well with metric deletion.
@shuhui-liu introduced a logic to delete metrics for the deleted stores, which will reduce unused metrics reporting to InGraph, and this issue is not critical right now since the store deletion rate is low.

In the future, we may want to register some store deletion listener here to remove unused metrics from `metricValueMap` and unregister them from MetricsRepository.

Reply:
Sure, I think this could be a future work to register clean up metrics..but it might not be trivial change.